### DISCUSSION TITLE
Ibl3

### DISCUSSION THREAD

**Discussion by looooo** [peterdsharpe/AeroSandbox#115] [General]
Implement a fully 3D coupled viscous/inviscid method, compatible with triangular panels (a la > Drela's IBL3 approach detailed in his paper "Three-Dimensional Integral Boundary Layer Formulation for General Configurations"). Ideally, the trailing edge stagnation points will be automatically identified, and nothing more than a surface triangulation along with freestream conditions will be required to compute forces and moments.

Very interesting topic. Just read a few documents on this topic. I have to admit that i do not understand a lot. But having an open source implementation available for this would be very nice. Do you know if there is an implementation of ibl3 available somewhere?

**Comment by peterdsharpe** [comment]
It is a neat topic, and it would be a truly revolutionary capability for aerodynamic analysis. The most up-to-date work on the topic is available here, in a thesis by one of my colleagues: https://dspace.mit.edu/handle/1721.1/147502

A preliminary implementation is contained within [SANS](https://darmofal.mit.edu/software), a CFD code developed by the MIT Aerospace Computational Design Lab (ACDL). But IBL3 really demands a standalone implementation, since it's not a "typical" PDE problem: you have a hyperbolic PDE solved on a 2D manifold, and then an outer-flow PDE that (in the linearized subsonic case) is elliptic and solvable with a Green's function approach.

There are still a few open challenges (i.e., corner flows, like wing-fuselage intersections) for IBL3 methods, but by and large the major outstanding task is implementation into a clean, open-source formulation.
